Mobile Theme: do not display settings when module is inactive.

jeherve opened the pull request:
Fixes #13385
Fixes #7602

Matching Calypso PR: https://github.com/Automattic/wp-calypso/pull/37857

#### Changes proposed in this Pull Request:

This PR comes as a first step to deprecate the Mobile Theme feature. We'll start by hiding Mobile Theme Settings from the Jetpack UI when the feature is not active. Folks that still use the feature can access its settings and disable it. Folks who do not use it cannot turn it on via the UI.

The Mobile theme was developed 8 years ago, back when responsive themes weren't a thing. It isn't as useful today, and can even be harmful when one enables it on their site without realizing the consequences, as outlined in #7602.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?

* Internal references:
    - p1HpG7-84F-p2
    - p6TEKc-3iM-p2
    - p6TEKc-3jj-p2
    - p6TEKc-3l7-p2

#### Testing instructions:

* Run `yarn docker:wp jetpack module activate minileven` to enable the module on your site.
* Go to Jetpack > Settings > Writing.
* Notice that the Mobile Theme settings are there. 
* Note that there is now a mention of the upcoming deprecation in the card.
![image](https://user-images.githubusercontent.com/426388/71005927-cb14ed00-20e4-11ea-88ef-dc8a8f8c21ac.png)
* Disable the feature.
* The settings should turn grey and be disabled.
* The settings should also be gone from the old module list.
* When clicking on the "Learn more" link in the notice, the support doc should open in a new window and you should stop a Tracks event if you add `localStorage.setItem( 'debug', 'dops:analytics' );` to your browser's console.

#### Proposed changelog entry for your changes:

* Mobile Theme: disable settings when feature is inactive.

jeherve committed:
Mobile Theme: do not display settings when module is inactive.

Fixes #13385
Fixes #7602

ChaosExAnima commented:
Code works, but the transition is very abrupt- it immediately vanishes when I click off. Can we have it so that it's there till the user refreshes the page, perhaps?

keoshi reviewed (dismissed):
This works as expected in my testing!

jeherve commented:
> Code works, but the transition is very abrupt- it immediately vanishes when I click off. Can we have it so that it's there till the user refreshes the page, perhaps?

That's a good point. I am not sure what the best approach would be for this, though. Should we fake the deactivation until they move to a different page, or tab?

dereksmart commented:
What if instead we just add a line of text to the card - something like: 

> This feature is being discontinued and will be removed from Jetpack in a future release. If you disable this setting, you will not be able to re-activate it.

jeherve committed:
Add mention of the future feature deprecation

See discussion here:
https://github.com/Automattic/jetpack/pull/14101#issuecomment-558233574

jeherve commented:
Done in afeb464aa781de9944aaf6f0f7542dab1398a3cb.

@jeffgolenski How does this look to you?

![image](https://user-images.githubusercontent.com/426388/69565272-f0667d80-0fb4-11ea-92a1-50f6f110c755.png)

You should be able to test on Jurassic Ninja as soon as the tests pass on this PR.  👍

jeffgolenski commented:
@jeherve That looks good to me. I'd personally like to have a little more visual call-out for this, but I don't think it's worth adding extra code to a component just for this notice. 🚢

kraftbj commented:
I agree with @ChaosExAnima that the transition is harsh, but given the overall intent to remove the feature soon, I'm not going to block merging based on that.

jeherve commented:
Putting this on hold for now since we may display things a bit differently.

Reference: p6TEKc-3iM-p2

jeherve committed:
Always display the Settings UI, but disable when module is off

Also use standard notice component to display notice.

jeherve committed:
Add new optional className to SettingsGroup component

jeherve committed:
Dynamically set minileven card class based on module status

jeherve committed:
Add basic styling when the Mobile theme feature is inactive

jeherve commented:
@jeffgolenski I'd love to have your opinion on that second pass. I've moved the notice to a notice component so things may be a bit easier to style and consistent with the rest of the dashboard. I added classes so inactive and active states can look different, but did not add any kind of CSS transition when moving from active to inactive.

<img width="1084" alt="screenshot 2019-12-09 at 15 29 26" src="https://user-images.githubusercontent.com/426388/70443732-ef474d00-1a98-11ea-9513-c65eb32e741a.png">
<img width="1137" alt="screenshot 2019-12-09 at 15 25 13" src="https://user-images.githubusercontent.com/426388/70443733-ef474d00-1a98-11ea-9869-71c012442985.png">

If you can work your magic there, I'd appreciate it :)

I won't touch the Calypso PR just yet, I'll wait until we're done with this one.

jeffgolenski commented:
@jeherve This is great. I prefer your variation since it uses existing components. Bravo! I'm just going to make some verbiage changes in my figma doc with this design and I think we'll be good to go. I'll post it here as soon as I'm done!

jeherve committed:
Update wording as per iteration 3

jeherve committed:
Add margin around notice, and transition when settings are greyed

jeffgolenski reviewed (approved):
Tested with both the theme already active, and inactive. Works great. Let's ship it.

kraftbj reviewed (approved):
Looks good. No errors in the console.
